src/test/java/io/vertx/rabbitmq/RabbitMQServiceTest.java

```diff
@@ -362,6 +367,42 @@ public void testGetMessageCount() throws Exception {
     await();
   }
 
+  @Test
+  public void consumerPrefetch(TestContext context) throws Exception {
+    // 1. Limit number of unack messages to 2
+    // 2. Send 3 messages
+    // 3. Ensure only 2 messages received
+    int count = 3;
+    int amountOfUnAckMessages = count - 1;
+
+    Async prefetchDone = context.async();
+    client.basicQos(amountOfUnAckMessages, done -> prefetchDone.countDown());
+    prefetchDone.await();
+
+    Set<String> messages = createMessages(count);
+    String queue = setupQueue(messages);
+    String address = queue + ".address";
+
+    Async receivedExpectedNumberOfMessages = context.async(amountOfUnAckMessages);
+
+    vertx.eventBus().consumer(address, msg -> {
+      if (receivedExpectedNumberOfMessages.isCompleted()) {
+        context.fail();
+      } else {
+        receivedExpectedNumberOfMessages.countDown();
+      }
+    });
+
+    client.basicConsume(queue, address, false, onSuccess(v -> {
+    }));
+
+    receivedExpectedNumberOfMessages.await();
+
+    // At the point we are sure, that we have already received 2 messages.
+    // But, if 3rd message will arrive the test will fail in the next second.
+    Thread.sleep(1000);
```

Comment:
not good, we should use Async#awaitSuccess instead

Reply:
I think it should rather be done with a vertx timer then, and the timer would complete an async after 1 second

Reply:
Done, can you take a look?